README.md

```diff
@@ -37,22 +44,147 @@ api("com.github.spotify:openfeature-kotlin-sdk:[ANY_BRANCH]-SNAPSHOT")
 
 This will get a build from the head of the mentioned branch. 
 
-### Resolving a flag
+## 🌟 Features
+
+- support for various backend [providers](https://openfeature.dev/docs/reference/concepts/provider)
+- easy integration and extension via [hooks](https://openfeature.dev/docs/reference/concepts/hooks)
+- bool, string, numeric, and object flag types
+- [context-aware](https://openfeature.dev/docs/reference/concepts/evaluation-context) evaluation
+
+## 🚀 Usage
+
+```kotlin
+    // configure a provider and get client
+    OpenFeatureAPI.setProvider(
+        CustomProvider.initialise()
+    )
+    val client = OpenFeatureAPI.getClient()
+
+    // get a bool flag value
+    client.getBooleanValue("boolFlag", default = false)
+    
+    // get a bool flag value async
+    openFeatureClient
```

Comment:
i wouldn't recommend this as primary usage of the client. instead, i'd recommend something like:
```
coroutineScope.launch {
WithContext(Dispatchers.IO) {
awaitProviderReady()
}
// read properties
}
```